Grbl_Esp32/src/Protocol.cpp

```diff
@@ -167,18 +167,49 @@ void protocol_main_loop() {
     int c;
     for (;;) {
         auto sdcard = config->_sdCard;
-        if (sdcard->_ready_next) {
+
+        // _readyNext means a line *might* be available from the _SDCard
+        // and implies SDCard::State == Busy (which is the same value as BusyPrinting).
```

Comment:
Actually, _ready_next means that the GCode processor has processed the previous line so it is appropriate to get another one.  The check for SDCard::State::BusyPrinting in report_status_message() controls whether to send an "ok" to a serial-ish sender vs. asking the SDCard for the next line.